runner/src/runner.rs

```diff
@@ -0,0 +1,1127 @@
+use crate::{error::Error, Opts};
+use backoff::{retry, Error as BackoffError, ExponentialBackoff};
+use bytes::Bytes;
+use codec::Decode;
+use futures::{future::BoxFuture, FutureExt, StreamExt, TryFutureExt};
+use nix::{
+	sys::signal::{self, Signal},
+	unistd::Pid,
+};
+use reqwest::Url;
+use sha2::{Digest, Sha256};
+use signal_hook_tokio::SignalsInfo;
+use sp_core::{hexdisplay::AsBytesRef, H256};
+use std::{
+	convert::TryInto,
+	fmt::{Debug, Display},
+	fs::{self, OpenOptions},
+	io::{Read, Seek, SeekFrom, Write},
+	os::unix::prelude::OpenOptionsExt,
+	path::PathBuf,
+	process::{Child, Command, Stdio},
+	str::{self, FromStr},
+	time::Duration,
+};
+
+use subxt::{dynamic::Value, OnlineClient, PolkadotConfig};
+
+use async_trait::async_trait;
+
+/// Type of the client to run.
+/// Also used as the name of the downloaded executable.
+#[derive(Debug, Clone)]
+pub enum ClientType {
+	Vault
+}
+
+impl FromStr for ClientType {
+	type Err = crate::Error;
+
+	fn from_str(s: &str) -> Result<Self, Self::Err> {
+		let client_type = match s {
+			"vault" => ClientType::Vault,
+			_ => return Err(Error::ClientTypeParsingError),
+		};
+		Ok(client_type)
+	}
+}
+
+impl Display for ClientType {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		let s = match self {
+			ClientType::Vault => "vault",
+		};
+		write!(f, "{s}")
+	}
+}
+
+/// Pallet in the parachain where the client release is assumed to be stored
+pub const PARACHAIN_MODULE: &str = "ClientInfo";
```

Comment:
I assume this matters.
```suggestion
pub const PARACHAIN_MODULE: &str = "ClientsInfo";
```

Reply:
It does!!